### PR TITLE
Update popup was not coming on JIRA for already exported BF

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Code Summary\Comments are added to my code which explains what my code is doing
- [x] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [x] Verify that changes are compatible with all relevant browsers and platforms.
- [x] After creating pull request there should not be any conflicts
- [x] Resolve all Codacy comments
- [x] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
  - Enhanced user messaging and null checks for better handling of business flows and activity groups during ALM exports.
  - Improved handling of user selections and updates to external IDs when exporting business flows to ALM.
  
- **Refactor**
  - Streamlined logic for exporting business flows to ALM.

### CODE REVIEW

File: Ginger/Ginger/ALM/Repository/JIRA_Repository.cs [hunk 1]
```diff
@@ -130,76 +131,120 @@ public override void ExportBfActivitiesGroupsToALM(BusinessFlow businessFlow, Ob
...
         public override bool ExportBusinessFlowToALM(BusinessFlow businessFlow, bool performSaveAfterExport = false, eALMConnectType almConectStyle = eALMConnectType.Manual, string testPlanUploadPath = null, string testLabUploadPath = null)
         {
+            if (businessFlow == null)
+            {
+                return false;
+            }
+
+            if (businessFlow.ActivitiesGroups.Count == 0 && almConectStyle != eALMConnectType.Silence)
+            {
+                Reporter.ToUser(eUserMsgKey.StaticInfoMessage, "The " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow) + " do not include " + GingerDicser.GetTermResValue(eTermResKey.ActivitiesGroups) + " which supposed to be mapped to ALM Test Cases, please add at least one " + GingerDicser.GetTermResValue(eTermResKey.ActivitiesGroup) + " before doing export.");
```
Comment (Maheshkale447) [inline Ginger/Ginger/ALM/Repository/JIRA_Repository.cs:141]: Use string interpolation

Comment (Maheshkale447) [inline Ginger/Ginger/ALM/Repository/JIRA_Repository.cs:141]: Add it in log as well

File: Ginger/Ginger/ALM/Repository/JIRA_Repository.cs [hunk 2]
```diff
@@ -130,76 +131,120 @@ public override void ExportBfActivitiesGroupsToALM(BusinessFlow businessFlow, Ob
...
-                    ObservableList<ExternalItemFieldBase> allFields = new ObservableList<ExternalItemFieldBase>(WorkSpace.Instance.Solution.ExternalItemsFields);
-                    ALMIntegration.Instance.RefreshALMItemFields(allFields, true, null);
-                    var testCaseFields = allFields.Where(a => a.ItemType == (ResourceType.TEST_CASE.ToString()) && (a.ToUpdate || a.Mandatory));
-                    var testSetFields = allFields.Where(a => a.ItemType == (ResourceType.TEST_SET.ToString()) && (a.ToUpdate || a.Mandatory));
-                    var testExecutionFields = allFields.Where(a => a.ItemType == "TEST_EXECUTION" && (a.ToUpdate || a.Mandatory));
-                    Reporter.ToStatus(eStatusMsgKey.ExportItemToALM, null, businessFlow.Name);
-                    bool exportRes = false;
-                    switch (ALMCore.DefaultAlmConfig.JiraTestingALM)
+                    //ask user if want to continute
+                    userSelec = Reporter.ToUser(eUserMsgKey.BusinessFlowAlreadyMappedToTC, businessFlow.Name, matchingTS.Name);
```
Comment (Maheshkale447) [inline Ginger/Ginger/ALM/Repository/JIRA_Repository.cs:156]: add if condition for silence

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/ALM/Repository/JIRA_Repository.cs [hunk 3]
```diff
@@ -130,76 +131,113 @@
...
+                }
+                return true;
+            }
+            else if (almConectStyle != eALMConnectType.Auto && almConectStyle != eALMConnectType.Silence)
+            {
+                Reporter.ToUser(eUserMsgKey.ExportItemToALMFailed, GingerDicser.GetTermResValue(eTermResKey.BusinessFlow), businessFlow.Name, responseStr);
             }
+
+            Reporter.HideStatusMessage();
+
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/Ginger/ALM/Repository/JIRA_Repository.cs:247]: handled

File: Ginger/Ginger/ALM/Repository/JIRA_Repository.cs (hunk 2 above)
Comment (AmanPrasad43) [reply Ginger/Ginger/ALM/Repository/JIRA_Repository.cs:156]: handled

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**